Calculator/CalculatorBrain.swift

```diff
@@ -18,14 +21,61 @@ class CalculatorBrain {
     var operatorType: OperatorType?
     
     func addOperandDigit(_ digit: String) -> String {
-        
+        guard operatorType == nil else {
+            operand2String.append(digit)
+            return operand2String
+        }
+        operand1String.append(digit)
+        return operand1String
     }
     
     func setOperator(_ operatorString: String) {
-        
+
+        switch operatorString {
+        case OperatorType.addition.rawValue:
+            operatorType = .addition
+        case OperatorType.subtraction.rawValue:
+            operatorType = .subtraction
+        case OperatorType.multiplication.rawValue:
+        operatorType = .multiplication
+        case OperatorType.division.rawValue:
+            operatorType = .division
+        default:
+            print("Error setting operator")
+        }
     }
     
     func calculateIfPossible() -> String? {
+        var answer: Double = 0.00
+        
+        guard operand2String != "",
+            operand1String != "",
+            operatorType != nil else {
+                print("No 1st/2nd operand or operator type")
+            return nil
+        }
+      guard let operand1 = Double(operand1String),
+            let operand2 = Double(operand2String),
+            Double(operand2String) != 0 else {
+            
+            return "Error"
+        }
         
+        switch operatorType {
+        case .addition:
+            answer = operand1 + operand2
+        case .subtraction:
+            answer = operand1 - operand2
+        case .multiplication:
+            answer = operand1 * operand2
+        case .division:
+            answer = operand1 / operand2
+        default:
+            return nil
+        }
+        
+     return String(answer)
     }
+    
```

Comment:
There is a more direct way to write all of this but this way works as well. For instance, all of your guard statements could have all been written together with an `if`.  Your `switch` statement would then all be part of that conditional. Like this. 

```
if !operand1String.isEmpty,
            !operand2String.isEmpty,
            let type = type,
            let operand1 = Double(operand1String),
            let operand2 = Double(operand2String) {

            switch type {
            case .addition:
                return String(operand1 + operand2)
            case .subtraction:
                return String(operand1 - operand2)
            case .multiplication:
                return String(operand1 * operand2)
            case .division:
                if operand2 != 0 {
                    return String(operand1 / operand2)
                } else {
                    return "Error"
                }
            }
        }
        return nil
    }
```